sycl/doc/extensions/proposed/sycl_ext_oneapi_record_event.asciidoc

```diff
@@ -0,0 +1,219 @@
+= sycl_ext_oneapi_record_event
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:endnote: &#8212;{nbsp}end{nbsp}note
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.
+Interfaces defined in this specification may not be implemented yet or may be
+in a preliminary state.
+The specification itself may also change in incompatible ways before it is
+finalized.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Overview
+
+This extension provides the ability to time the execution of commands in a
+queue without enabling profiling on the entire queue.
+This is more efficient on some platforms because only a subset of the events
+are required to contain timestamp information.
+It is also more convenient for use in libraries, where the library wants to
```

Comment:
Yes, I think that some backends might have this limitation.  In fact, OpenCL is like this now.  Unless we add an extension to OpenCL, an OpenCL backend wouldn't be able to support `record_event` without enabling queue profiling by default.

This is why I made the feature optional and tied it to a new aspect named `ext_oneapi_queue_event_recording`.  I expect backends that have this limitation will return false for that aspect, which means `record_event` won't work for those backends.

Reply:
That's fair, having this be an optional feature means that implementations/backends can opt-out of it's felt that it can't be implemented efficiently.